test/integration/docker/image.py

```diff
@@ -0,0 +1,66 @@
+import json
+import re
+from datetime import datetime
+from typing import List, Dict, Any, Tuple, Optional
+
+import docker
+from docker.errors import BuildError
+from docker.models.images import Image
+
+
+def format_build_log(build_log: List[Dict[str, Any]]):
+    def format_entry(entry: Dict[str, Any]):
+        if "stream" in entry:
+            return entry["stream"]
+        if "error" in entry:
+            return entry["error"]
+        return ""
+
+    return "\n".join(format_entry(entry) for entry in build_log)
+
+
+class BuildErrorWithLog(BuildError):
+    def __init__(self, reason, build_log: List[Dict[str, Any]]):
+        super().__init__(f"{reason}\n\n{format_build_log(build_log)}", build_log)
+
+
+def image(request, name: str, print_log=False, **kwargs) -> Image:
```

Comment:
```suggestion
def image(request, name: str, print_log=False, **kwargs) -> Image:
    """
    Create a Docker image.
    The function supports a pair of pytest cli options with a suffix derived from parameter ``name``:
    Option `--docker-image-(suffix)` specifies the name of an existing image to be used 
    instead of creating a new one.
    Option `--keep-docker-image-(suffix)` skips removing the image after test execution.
    """
```